recipes_source/recipes/profiler_recipe.py

```diff
@@ -197,6 +198,46 @@
 #    Self CPU time total: 23.015ms
 #    Self CUDA time total: 11.666ms
 #
+######################################################################
+# Profiler can also be used to analyze performance of models executed on XPUs:
+
+model = models.resnet18().xpu()
```

Comment:
That's python script, not rst document. As such, mixing up explicit examples for cuda and xpu would give issues to people to try this example since they might not have both cuda and xpu on the system. So, they will start seeing errors.

Maybe it's better to follow the pattern to abstract device with `.to(device)`, hardcode `cuda` and let user modify for `xpu`?

Reply:
Yes, we should refactor this python script to be device-agnostic and let the user know it supports on cuda and xpu. And in the future, the other backend also can expand it.Yes, we should refactor this python script to be device-agnostic and let user know it supports on cuda and xpu. And in the future, the other backend also can expand it.

Reply:
some goods. Let me work on that and update PR later this week.